zjit/src/stats.rs

```diff
@@ -320,6 +321,8 @@ pub fn exit_counter_for_compile_error(compile_error: &CompileError) -> Counter {
                 JumpTargetNotInRPO(_)         => compile_error_validation_jump_target_not_in_rpo,
                 OperandNotDefined(_, _, _)    => compile_error_validation_operand_not_defined,
                 DuplicateInstruction(_, _)    => compile_error_validation_duplicate_instruction,
+                MismatchedOperandType(..)     => compile_error_validation_type_check_failure,
+                MiscValidationError(..)       => compile_error_validation_type_check_failure,
```

Comment:
Can we give different counter names to `MismatchedOperandType` and `MiscValidationError`? I get that both of them are type check failures, but it's generally nice to at least know which enum variant it was when you actually work on it.

Reply:
I'll split the counters for this PR though